src/search/heuristics/additive_heuristic.cc

```diff
@@ -18,15 +18,17 @@ const int AdditiveHeuristic::MAX_COST_VALUE;
 AdditiveHeuristic::AdditiveHeuristic(const Options &opts)
     : RelaxationHeuristic(opts),
       did_write_overflow_warning(false) {
-    utils::g_log << "Initializing additive heuristic..." << endl;
+    if (log.is_at_least_normal()) {
+        log << "Initializing additive heuristic..." << endl;
+    }
 }
 
 void AdditiveHeuristic::write_overflow_warning() {
     if (!did_write_overflow_warning) {
         // TODO: Should have a planner-wide warning mechanism to handle
         // things like this.
-        utils::g_log << "WARNING: overflow on h^add! Costs clamped to "
-                     << MAX_COST_VALUE << endl;
+        log << "WARNING: overflow on h^add! Costs clamped to "
```

Comment:
Yes. I'm not sure I agree it is useful to do this now, but fine by me. What is weird, though, is that here we have a warning and "a warning on stderr" right below. Should we remove one of these?

Reply:
The motivation is that we want to have it on `cerr` so that it shows up (for example) in the "unexpected errors" of the lab report, but we also want it to show up in the log (which it wouldn't if it were only printed on `cerr`). I'd leave this as-is for now, but it's something that we could address in a better way by improving our logging framework. For example, warning-level (or worse) log messages could *always* be printed on both streams, or this could be configurable.

Reply:
I thought our convention was to exit after printing on `cerr`.